tor_core/initialize.py

```diff
@@ -211,17 +213,67 @@ def initialize(config):
     logging.debug('Gifs loaded.')
 
 
-def build_bot(name, log_name='transcribersofreddit.log', require_redis=True):
+def get_heartbeat_port(config):
+    """
+    Attempts to pull an existing port number from the filesystem, and if it
+    doesn't find one then it generates the port number and saves it to a key
+    file.
+
+    :param config: the global config object
+    :return: int; the port number to use.
+    """
+    try:
+        # have we already reserved a port for this process?
+        port = int(open('heartbeat.port', 'r').readline().strip())
+        logging.debug('Found existing port saved on disk')
+        return port
+    except OSError:
+        pass
+
+    while True:
+        port = random.randrange(40000, 40200)  # is 200 ports too much?
+        if config.redis.sismember('active_heartbeat_ports', port) == 0:
+            config.redis.sadd('active_heartbeat_ports', port)
+
+            # create that file we looked for earlier
+            port_file = open('heartbeat.port', 'w')
+            port_file.write(str(port))
+            port_file.close()
```

Comment:
What about shortening this using `with`? Then, in case you do more while the file is open, the `port_file.close()` line going missing won't create something like a memory leak.

```python
with open('heartbeat.port', 'w') as port_file:
    port_file.write(str(port))
```